### PR TITLE
Add support for `setup` steps in flutter customer tests

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR adds support for adding `setup` steps that must be run before the `update` steps in Flutter customer tests.

```
setup=./generate_my_mocks.sh
update=.
test= ...
```

This PR also cleans up the `CustomerTest` class in the spirit of yak-shaving and making the code more robust.

### CODE REVIEW

File: dev/customer_testing/lib/customer_test.dart [hunk 1]
```diff
@@ -106,7 +110,60 @@ class CustomerTest {
...
+        _PlatformType.macos => Platform.isMacOS,
+        _PlatformType.linux => Platform.isLinux,
+        _PlatformType.posix => Platform.isLinux || Platform.isMacOS,
+      };
+}
+
+enum _TestDirective {
+  contact,
+  fetch,
+  setup,
```
Comment (goderbauer) [inline dev/customer_testing/lib/customer_test.dart:165]: Can you please also update https://github.com/flutter/tests/blob/main/registry/template.test to mention and document the new directive? Thank you!

**Review by goderbauer — APPROVED** [review]
LGTM

File: dev/customer_testing/lib/customer_test.dart [hunk 2]
```diff
@@ -12,92 +13,95 @@ class CustomerTest {
...
-      } else if (line.startsWith('iterations=')) {
+      if (line.isEmpty || line.startsWith('#')) {
+        // blank line or comment
+        continue;
+      }
+
+      final bool isUnknownDirective = _TestDirective.values.firstWhereOrNull(
+            (_TestDirective d) => line.startsWith(d.name),
+          ) ==
+          null;
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply dev/customer_testing/lib/customer_test.dart:30]: done

File: dev/customer_testing/lib/customer_test.dart [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 import 'dart:io';
 
+import 'package:collection/collection.dart';
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply dev/customer_testing/lib/customer_test.dart:7]: `firstWhereOrNull` is from `package:collection`

File: dev/customer_testing/lib/customer_test.dart (hunk 1 above)
Comment (kenzieschmoll) [reply dev/customer_testing/lib/customer_test.dart:165]: Done in this PR https://github.com/flutter/tests/pull/267

**Pull request merged**